cloud_experts_tutorials/cloud-experts-dynamic-certificate-custom-domain.adoc

```diff
@@ -0,0 +1,522 @@
++
+[source,terminal]
+----
+$ oc -n cert-manager apply -f https://github.com/cert-manager/openshift-routes/releases/latest/download/cert-manager-openshift-routes.yaml
+----
++
+Additional OpenShift resources such as a `ClusterRole` (with permissions to watch and update the routes across the cluster), a `ServiceAccount` (with these permissions, that will be used to run this newly created pod), and a `ClusterRoleBinding` to bind these two resources, will be created in this step too.
++
+[NOTE]
+====
+If the cluster does not have access to github, you can save the raw contents locally, and run `oc apply -f localfilename.yaml -n cert-manager`.
+====
+. Check to ensure that the new pod `cert-manager-openshift-routes` is running successfully by running the following command:
++
+[source,terminal]
+----
+$ oc -n cert-manager get pods
+----
++
+.Example result
++
+[source,terminal]
+----
+NAME                                             READY   STATUS    RESTARTS   AGE
+cert-manager-866d8f788c-9kspc                    1/1     Running   0          4h21m
+cert-manager-cainjector-6885c585bd-znws8         1/1     Running   0          4h41m
+cert-manager-openshift-routes-75b6bb44cd-f8kd5   1/1     Running   0          6s
+cert-manager-webhook-8498785dd9-bvfdf            1/1     Running   0          4h41m
+----
+
+== Deploy a sample application
+
+. Create a new project for our sample application:
++
+[source,terminal]
+----
+$ oc new-project hello-world
+----
++
+. Deploy a hello world application: 
++
+[source,terminal]
+----
+$ oc -n hello-world new-app --image=docker.io/openshift/hello-openshift
+----
++
+. Create a route to expose the application from outside the cluster:
++
+[source,terminal]
+----
+$ oc -n hello-world create route edge --service=hello-openshift hello-openshift-tls --hostname hello.${DOMAIN}
+----
++
+. Verify the certificate for the route is untrusted:
++
+[source,terminal]
+----
+$ curl -I https://hello.${DOMAIN}
+----
+.Example output
++
+[source,text]
+----
+curl: (60) SSL: no alternative certificate subject name matches target host name 'hello.example.com'
+More details here: https://curl.se/docs/sslcerts.html
+
+curl failed to verify the legitimacy of the server and therefore could not
+establish a secure connection to it. To learn more about this situation and
+how to fix it, please visit the web page mentioned above.
+----
++
+. Annotate the route to trigger cert-manager to provision a certificate for the custom domain: 
++
+[source,terminal]
+----
+$ oc -n hello-world annotate route hello-openshift-tls cert-manager.io/issuer-kind=ClusterIssuer cert-manager.io/issuer-name=letsencrypt-production
+----
++
+[NOTE]
+====
+It will take a 2-3 minutes for the certificate to be created. The renewal of the certificate will automatically be managed by the cert-manager Operator as it approaches expiry.
```

Comment:
```suggestion
It will take a 2-3 minutes for the certificate to be created. The renewal of the certificate will automatically be managed by the cert-manager Operator as it approaches expiration.
```